pkgs/applications/graphics/drawpile/default.nix

```diff
@@ -25,24 +34,47 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [
     extra-cmake-modules
   ];
-  buildInputs = [
-    # common deps:
+  buildInputs =
+    commonDeps ++
+    optionals buildClient     (clientDeps ++ clientOptionalDeps) ++
+    optionals buildServer     (serverDeps ++ serverOptionalDeps) ++
+    optionals enableKisTablet  kisDeps                           ;
+  commonDeps = [
     cmake
-    qtbase qtsvg qtmultimedia qttools
     karchive
-    # optional deps:
-    #   server-specific:
-    libsodium # ext-auth support
+  ];
+  clientDeps = [
```

Comment:
You unconditionally pass `clientDeps`, `serverDeps`, `kisDeps`, `serverOptionalDeps` and `clientOptionalDeps` as attributes to derivation which effectively makes all of the possible dependencies required to be built before building this derivation.

Reply:
Oops, yeah this should be outside of that set, thanks.